pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -20,15 +20,21 @@ import (
 	"net/http"
 
 	genericapifilters "k8s.io/apiserver/pkg/endpoints/filters"
+	"k8s.io/apiserver/pkg/features"
 	"k8s.io/apiserver/pkg/server"
 	genericfilters "k8s.io/apiserver/pkg/server/filters"
+	"k8s.io/apiserver/pkg/util/feature"
 )
 
 // BuildInsecureHandlerChain sets up the server to listen to http. Should be removed.
 func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.Handler {
 	handler := apiHandler
-	handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.LongRunningFunc)
 	handler = genericapifilters.WithAudit(handler, c.AuditBackend, c.AuditPolicyChecker, c.LongRunningFunc)
+	if feature.DefaultFeatureGate.Enabled(features.RequestManagement) {
```

Comment:
I'd rather not wire this into the insecure handler.  insecure handling is legacy and maybe feature gaps will help move people.

Reply:
If you want to break insecure handling, why be subtle about it?  I see no point in going half-way; either we maintain it or delete it.